### PR TITLE
Support CocoaPods

### PR DESCRIPTION
**Pull request by simorgh3196**
Issue: #52 

I hope I can install this library with CocoaPods.

I tried installing with CocoaPods by referring to `SwiftLint` etc.
I did not actually register with CocoaPods, so I can not confirm whether it will work, but I tried making a PR wishing this would be useful.

I would be happy if you can provide feedback if there are still missing operations.

## TODO
- [ ] Register CocoaPods
- [ ] Installation Test

### CODE REVIEW

**Comment by mono0926** [comment]
Thanks, very much!
It seems to be great.
I have never supported CocoaPods distribution, so I postponed #52. Ando so, this PR is very helpful 😋 

> ## TODO
> - [ ] Register CocoaPods
> - [ ] Installation Test

How should I treat this TODO?

Do you demand me to merge to master to deal with the them?
If so, I'll merge this.

In advance, I noticed a review point, so I'll comment now.

File: Makefile [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+
 install: build
 	mkdir -p "$(PREFIX)/bin"
 	cp -f ".build/release/LicensePlist" "$(PREFIX)/bin/license-plist"
+
+portable_zip: build
+	mkdir "$(TEMPORARY_FOLDER)"
+	cp -f ".build/release/LicensePlist" "$(TEMPORARY_FOLDER)"
+	cp -f "LICENSE" "$(TEMPORARY_FOLDER)"
+	(cd tmp/portable_licenseplist; zip -r - LICENSE LicensePlist) > './portable_licenseplist.zip'
```
Comment (mono0926) [inline Makefile]: Is `tmp/portable_licenseplist` correct?
It seems to be that `$(TEMPORARY_FOLDER)`(`tmp_portable_licenseplist `) is correct for me.

File: Makefile [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 clean:
 	swift package clean
 
 xcode:
 	swift package generate-xcodeproj
-	
+
 install: build
 	mkdir -p "$(PREFIX)/bin"
 	cp -f ".build/release/LicensePlist" "$(PREFIX)/bin/license-plist"
```
Comment (mono0926) [inline Makefile]: The executable binary is renamed to chain case from camel case here when installed via Makefile or Homebrew.
It is better to also apply it `cp -f ".build/release/LicensePlist" "$(TEMPORARY_FOLDER)"`, isn't it?
(I don't understand CocoaPods distribution well.)

File: Makefile [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
+
+portable_zip: build
+	mkdir "$(TEMPORARY_FOLDER)"
+	cp -f ".build/release/LicensePlist" "$(TEMPORARY_FOLDER)"
```
Comment (mono0926) [inline Makefile]: Should be `cp -f ".build/release/LicensePlist" "$(TEMPORARY_FOLDER)/license-plist"`?

File: README.md [hunk 4]
```diff
@@ -138,6 +144,15 @@ fi
 
 ![](Screenshots/pre_build_action.png)
 
+Alternatively, if you've installed LicensePlist via CocoaPods the script should look like this:
+
+```sh
+if [ $CONFIGURATION = "Debug" ]; then
+cd $SRCROOT
+${PODS_ROOT}/LicensePlist/portable_licenseplist --output-path $PRODUCT_NAME/Settings.bundle
```
Comment (mono0926) [inline README.md]: I the case of [SwiftLint](https://github.com/realm/SwiftLint) CocoaPods distribution, its executable binary is located at `Pods/SwiftLint/swiftlint`.

So, I think `${PODS_ROOT}/LicensePlist/license-plist --output-path $PRODUCT_NAME/Settings.bundle` is more natural.
(Of course, it is necessary to modify other scripts to fit this command if needed.)

File: LicensePlist.podspec.tmp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Pod::Spec.new do |s|
+  s.name            = 'LicensePlist'
+  s.version         = 'LATEST_RELEASE_VERSION_NUMBER'
+  s.summary         = 'A license list generator of all your dependencies for iOS applications'
+  s.homepage        = 'https://github.com/mono0926/LicensePlist'
+  s.license         = { :type => 'MIT', :file => 'LICENSE' }
+  s.author          = 'Masayuki Ono'
+  s.source          = { :http => '#{s.homepage}/releases/download/#{s.version}/portable_licenseplist.zip' }
+end
```
Comment (mono0926) [inline LicensePlist.podspec.tmp]: Why do this file name contain `.tmp`?
I think you can test CocoaPods installation with this command by removing `.tmp`.

```ruby
pod 'LicensePlist', :git => 'https://github.com/simorgh3196/LicensePlist.git', :branch => 'cocoapods' 
```

Comment (mono0926) [inline LicensePlist.podspec.tmp]: > `s.source          = { :http => '#{s.homepage}/releases/download/#{s.version}/portable_licenseplist.zip' }`

I noticed that, it is required to modify this for testing temporarily.
If it is difficult to test in advance, I'll merge it after the review requests is processed.

File: README.md (hunk 4 above)
Comment (simorgh3196) [inline README.md]: Oops. I rename binary name to `license-plist`, also modify others.

File: LicensePlist.podspec.tmp (hunk 5 above)
Comment (simorgh3196) [inline LicensePlist.podspec.tmp]: In order to deliver CocoaPods updates in release.sh, I wanted to be able to set the version number externally.

Therefore, I thought that create a new podspec dynamicary, and replace version number.

> LicensePlist.podspec.tmp
> ```ruby
> s.version = 'LATEST_RELEASE_VERSION_NUMBER'
> ```

> release.sh
> ```bash
> cat "$podspec_name.tmp" | sed s/LATEST_RELEASE_VERSION_NUMBER/$tag/ > "$podspec_name"
> ```
If you have any other good ideas please let me know.

Comment (mono0926) [inline LicensePlist.podspec.tmp]: Okay, I understand. It seems to be good 👍 

(SwiftLint use [another technique](https://github.com/realm/SwiftLint/blob/fd90858e4a7d9566749975fb5133f9276444721b/SwiftLint.podspec#L3) though.)

**Comment by mono0926** [comment]
Thank you for the PR.

I noticed that you tested the installation: https://github.com/simorgh3196/LicensePlist/tree/cocoapods-test 👍 

After I merged this to master, I executed `./release.sh 1.6.0 GITHUB_TOKEN` but it failed.
So I reverted README temporarily.
See this commit message: https://github.com/mono0926/LicensePlist/commit/f30208f5a08d1a4e7a9446a2d44858e39314250d

If you could resolve this error, submit a new PR.
Or I'll try to resolve it later.

**Pull request merged**